src/workflows-app/SubmissionDetails.js

```diff
@@ -140,8 +141,11 @@ export const BaseSubmissionDetails = ({ name, namespace, workspace, submissionId
       if (workflowsAppStore.get().cbasProxyUrlState.status === AppProxyUrlStatus.Ready) {
         updatedRunSets = await loadAllRunSets(workflowsAppStore.get().cbasProxyUrlState);
       }
-      // only refresh if there are run sets in non-terminal state
-      if (!updatedRunSets || _.some(({ state }) => !isRunSetInTerminalState(state), updatedRunSets)) {
+      // only refresh if _this_ run set is in non-terminal state
```

Comment:
This was the primary issue

Reply:
Why are we loading _all_ run sets to display a singe submission's details?

Reply:
We don't have an endpoint to do this currently. I talked to @salonishah11 and she mentioned there were plans to implement it at some point, but I wasn't sure whether there would be other changes involved.